pkg/gitrepo/repo_test.go

```diff
@@ -393,53 +394,58 @@ func Test_Repo_GetFileContent(t *testing.T) {
 			expected: "DCO",
 		},
 		{
-			name:     "case 1: get DCO file content on branch-of-2.0.0 branch",
+			name:     "case 1: get DCO file content on default branch (master)",
+			path:     "DCO",
+			expected: "DCO",
+			ref:      "master",
+		},
+		{
+			name:     "case 2: get DCO file content on branch-of-2.0.0 branch",
 			path:     "DCO",
 			expected: "DCO",
 			ref:      "origin/branch-of-2.0.0",
 		},
 		{
-			name:     "case 2: get DCO file content on v2.0.0 tag",
+			name:     "case 3: get DCO file content on v2.0.0 tag",
 			path:     "DCO",
 			expected: "DCO",
 			ref:      "v2.0.0",
 		},
 		{
-			name:         "case 3: handle file not found error",
+			name:         "case 4: handle file not found error",
 			path:         "non/existent/file/path",
 			errorMatcher: IsFileNotFound,
 		},
 		{
-			name:         "case 4: handle reference not found error",
+			name:         "case 5: handle reference not found error",
 			path:         "DCO",
 			ref:          "does-not-exist",
 			errorMatcher: IsReferenceNotFound,
 		},
 	}
 
-	dir := "/tmp/gitrepo-test-repo-getfilecontent"
-	defer os.RemoveAll(dir)
-
-	c := Config{
-		Dir: dir,
-		URL: "git@github.com:giantswarm/gitrepo-test.git",
-	}
-	repo, err := New(c)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	ctx := context.Background()
-
-	err = repo.EnsureUpToDate(ctx)
-	if err != nil {
-		t.Fatal(err)
-	}
-
 	for i, tc := range testCases {
 		t.Run(strconv.Itoa(i), func(t *testing.T) {
 			t.Log(tc.name)
 
+			dir := fmt.Sprintf("/tmp/gitrepo-test-repo-getfilecontent-%d", i)
```

Comment:
Start clean for each test so that they don't fail/succeed because of leftovers of previous tests.

Reply:
I don't like this has a potential to leave a mess on the OS. Why ensuring removal of the directory as a first operation here?

Reply:
It's cleaning things up in a defer so those directories should always get cleaned up. If you ensure the removal at a later point, there is the risk of something failing and leaving those dirs behind. Also I haven't really introduced new behaviour, I just moved it to each individual test so that the folder doesn't get reused between them.

Reply:
I won’t argue about that. It’s already approved.